src/widget-sources/widget-base-source.ts

```diff
@@ -93,14 +94,35 @@ export abstract class WidgetBaseSource<Props extends WidgetBaseSourceProps> {
   async getCategories(
     options: CategoryRequestOptions
   ): Promise<CategoryResponse> {
-    const {filterOwner, spatialFilter, abortController, ...params} = options;
+    const {
+      filterOwner,
+      spatialFilter,
+      spatialFiltersMode,
+      abortController,
+      viewState,
+      ...params
+    } = options;
     const {column, operation, operationColumn} = params;
+    const source = this.getModelSource(filterOwner);
+
+    let spatialFiltersResolution = options.spatialFiltersResolution;
+    if (spatialFilter && !spatialFiltersResolution) {
+      spatialFiltersResolution = getSpatialFiltersResolution({
+        source,
+        viewState,
+      });
+    }
```

Comment:
Following https://github.com/CartoDB/carto-api-client/pull/34#discussion_r1871356107, I think we also don't want to expose the `spatialFiltersResolution` option to users yet, so we can unconditionally compute it here, if a spatial filter is used.

I'm not opposed to adding it later, but as Felix mentions there is some subtlety about how we can/should do that, and it doesn't seem essential to get this feature shipped now.

Reply:
Ok I'm merging this comment with some other from my team in PS to take the early return out of the `getSpatialFiltersResolution` and add it the conditional there